wrapper/src/main/java/software/amazon/jdbc/ConnectionPluginChainBuilder.java

```diff
@@ -0,0 +1,229 @@
+          put(IamAuthConnectionPluginFactory.class, 900);
+          put(AwsSecretsManagerConnectionPluginFactory.class, 1000);
+          put(LogQueryConnectionPluginFactory.class, 1100);
+          put(ConnectTimeConnectionPluginFactory.class, STICK_TO_PRIOR_PLUGIN);
+          put(ExecutionTimeConnectionPluginFactory.class, STICK_TO_PRIOR_PLUGIN);
+          put(DeveloperConnectionPluginFactory.class, STICK_TO_PRIOR_PLUGIN);
+        }
+      };
+
+  protected static final String DEFAULT_PLUGINS = "auroraConnectionTracker,failover,efm";
+
+  private static class PluginFactoryInfo {
+    public Class<? extends ConnectionPluginFactory> factory;
+    public int weight;
+
+    public PluginFactoryInfo(final Class<? extends ConnectionPluginFactory> factory, final int weight) {
+      this.factory = factory;
+      this.weight = weight;
+    }
+  }
+
+  public List<ConnectionPlugin> getPlugins(
+      final PluginService pluginService,
+      final ConnectionProvider defaultConnProvider,
+      final PluginManagerService pluginManagerService,
+      final Properties props)
+      throws SQLException {
+
+    List<ConnectionPlugin> plugins;
+    List<Class<? extends ConnectionPluginFactory>> pluginFactories;
+
+    final String profileName = PropertyDefinition.PROFILE_NAME.getString(props);
+
+    if (profileName != null) {
+
+      if (!DriverConfigurationProfiles.contains(profileName)) {
+        throw new SQLException(
+            Messages.get(
+                "ConnectionPluginManager.configurationProfileNotFound",
+                new Object[] {profileName}));
+      }
+      pluginFactories = DriverConfigurationProfiles.getPluginFactories(profileName);
+
+    } else {
+
+      String pluginCodes = PropertyDefinition.PLUGINS.getString(props);
+
+      if (pluginCodes == null) {
+        pluginCodes = DEFAULT_PLUGINS;
+      }
+
+      final List<String> pluginCodeList = StringUtils.split(pluginCodes, ",", true);
+      pluginFactories = new ArrayList<>(pluginCodeList.size());
+
+      for (final String pluginCode : pluginCodeList) {
+        if (!pluginFactoriesByCode.containsKey(pluginCode)) {
+          throw new SQLException(
+              Messages.get(
+                  "ConnectionPluginManager.unknownPluginCode",
+                  new Object[] {pluginCode}));
+        }
+        pluginFactories.add(pluginFactoriesByCode.get(pluginCode));
+      }
+    }
+
+    if (!pluginFactories.isEmpty()) {
+
+      if (!PropertyDefinition.PRESERVE_PLUGINS_ORDER.getBoolean(props)) {
+        pluginFactories = this.sortPluginFactories(pluginFactories);
+
+        final List<Class<? extends ConnectionPluginFactory>> tempPluginFactories = pluginFactories;
+        LOGGER.finest(() ->
+            "Plugins order has been rearranged. The following order is in effect: "
+                + tempPluginFactories.stream()
+                  .map(Class::getSimpleName)
+                  .collect(Collectors.joining(", ")));
+      }
+
```

Comment:
is it worth to separate the code below into a function loadFactories() that returns the factories list?